diff --git a/mdloader_common.c b/mdloader_common.c
index bfe8b0d..a00f259 100644
--- a/mdloader_common.c
+++ b/mdloader_common.c
@@ -336,7 +336,7 @@ int test_mcu(char silent)
             continue;
         }
 
-        if (deviceid == mcu->cidr)
+        if ((deviceid & 0xFFFFF0FF) == mcu->cidr)
         {
             if (!silent && verbose) printf("Found supported device ID: %08X\n", deviceid);
             break;
@@ -728,10 +728,9 @@ int main(int argc, char *argv[])
 
     //Load applet
     FILE *fIn;
-    char appletfname[128] = "";
+    char appletfname[128] = "@out@/bin/applet-flash-samd51j18a.bin";
     strlower(mcu->name);
 
-    sprintf(appletfname, "applet-flash-%s.bin", mcu->name);
     printf("Applet file: %s\n", appletfname);
 
     fIn = fopen(appletfname, "rb");
